Taskfile.yml

```diff
@@ -100,10 +103,61 @@ tasks:
       NODE_PATH:
         sh: npm root -g
 
-  ############### Generator targets ###############
   basic-checks:
     deps: [header-check, specifier-check]
 
+  ############### Asoctl targets ###############
+
+  asoctl:quick-checks:
+    cmds:
+      - task: asoctl:unit-tests
+      - task: basic-checks
+      - task: asoctl:lint
+
+  asoctl:ci:
+    cmds:
+      - task: asoctl:unit-tests-cover
+      - task: basic-checks
+      - task: asoctl:lint-full
+
+  asoctl:unit-tests:
+    desc: Run {{.ASOCTL_APP}} unit tests.
+    dir: '{{.ASOCTL_ROOT}}'
+    cmds:
+      - go test ./... -tags=noexit -run '{{default ".*" .TEST_FILTER}}'
+
+  asoctl:unit-tests-cover:
+    desc: Run {{.ASOCTL_APP}} unit tests and output coverage.
+    dir: '{{.ASOCTL_ROOT}}'
+    cmds:
+      - defer: # want to run even on failure
+          task: produce-markdown-summary
+          vars:
+            INPUT_FILE: '{{.TEST_OUT}}/asoctl-unit-tests.json'
+            OUTPUT_FILE: '{{.TEST_OUT}}/asoctl-unit-tests.md'
+      - go test ./... -tags=noexit -race -covermode atomic -coverprofile=asoctl-coverage.out -json -coverpkg=./... -run '{{default ".*" .TEST_FILTER}}' > '{{.TEST_OUT}}/asoctl-unit-tests.json'
+
+
+  asoctl:lint:
+    desc: Run {{.ASOCTL_APP}} fast lint checks.
+    dir: '{{.ASOCTL_ROOT}}'
+    cmds:
+      - golangci-lint run --verbose
+
+  asoctl:lint-full:
```

Comment:
I know you followed the pattern here, but I'm thinking we actually should just remove all the `lint-full`'s and make `lint` use `--fast=false` everywhere. Otherwise CI and quick-checks might run different validations which feels awkward (passes locally, fails in CI).

I can fix that once this merges.